lib/rucio/common/stomp_utils.py

```diff
@@ -15,145 +15,415 @@
+                                        "and 'ssl_key_file'")
+            raise RuntimeError("If use_ssl is True in config you must provide both 'ssl_cert_file' and 'ssl_key_file'")
+        if not use_ssl and (username is None or password is None or nonssl_port == 0):
+            self._logger(logging.ERROR, "If use_ssl is False in config you must provide "
+                                        "'username', 'password' and 'nonssl_port'")
+            raise RuntimeError("If use_ssl is False in config you must provide "
+                               "'username', 'password' and 'nonssl_port'")
+        return StompConfig(brokers=self._resolve_host_and_port(brokers, port if use_ssl else nonssl_port),
+                           use_ssl=use_ssl,
+                           port=port, vhost=vhost,
+                           destination=destination, key_file=key_file, cert_file=cert_file,
+                           username=username, password=password, nonssl_port=nonssl_port,
+                           reconnect_attempts_max=reconnect_attempts, timeout=timeout, heartbeats=heartbeats)
+
+    def _resolve_host_and_port(self, fqdns: str | list[str], port: int) -> list[tuple[str, int]]:
+        """
+        Resolve host and port.
+
+        Args:
+            fqdns (str | list[str]): fully qualified domain name(s)
+            port (int): port
+
+        Returns:
+            list[tuple[str, int]]: list of (host, port) tuples.
+        """
+        if isinstance(fqdns, str):
+            fqdns = fqdns.split(',')
+
+        hosts_and_ports = []
+        for fqdn in fqdns:
+            try:
+                addrinfos = socket.getaddrinfo(fqdn.strip(), port, socket.AF_INET, 0, socket.IPPROTO_TCP)
+            except socket.gaierror as exc:
+                self._logger(logging.ERROR, "[broker] Cannot resolve domain name %s (%s)", fqdn.strip(), str(exc))
+                continue
+
+            hosts_and_ports.extend(addrinfo[4] for addrinfo in addrinfos)
+        if not hosts_and_ports:
+            self._logger(logging.WARNING, "[broker] No resolved brokers")
+        return hosts_and_ports
+
+    def _is_stalled(self, conn: Connection) -> bool:
+        """
+        Determine if a connection is stalled.
+
+        Args:
+            conn (Connection): The Connection object
+
+        Returns:
+            bool: Whether the connection has stalled.
+        """
+        received_heartbeat = getattr(conn, 'received_heartbeat', None)
+        if received_heartbeat is None or not any(self._config.heartbeats):
+            return False
+
+        heartbeat_period_seconds = max(0, self._config.heartbeats[0], self._config.heartbeats[1]) / 1000
+        if heartbeat_period_seconds == 0.:
+            return False
+
+        now = monotonic()
+        if received_heartbeat + 10 * heartbeat_period_seconds >= now:
+            return False
+
+        return True
+
+    def connect(self) -> Generator[Connection, None, None]:
```

Comment:
Can be simplified to:

```suggestion
    def connect(self) -> Iterator[Connection]:
```

(see best practices here: https://rucio.github.io/documentation/developer/type_annotation_guide/#best-practices)